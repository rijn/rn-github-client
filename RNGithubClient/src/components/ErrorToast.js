import { Toast } from 'native-base';

export default (e) => {
  Toast.show({
    text: e.message,
    position: 'top',
    buttonText: 'Okay',
    type: 'danger',
    duration: 3000
  });
}