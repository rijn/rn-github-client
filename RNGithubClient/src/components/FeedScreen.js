import React from 'react';
import { connect } from 'react-redux';
import { StyleSheet, View } from 'react-native';
import { Button, Text } from 'native-base';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f5f5f5',
  },
});

/**
 * Class for feed screen.
 *
 * @desc All user activities
 *
 * @class FeedScreen
 */
class FeedScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    let { dispatch } = this.props;

    return (
      <View style={styles.container}>
        <Text>Main</Text>
      </View>
    );
  }
};

export default connect()(FeedScreen);