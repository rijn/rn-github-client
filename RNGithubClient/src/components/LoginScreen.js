import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, AsyncStorage } from 'react-native';
import Entypo from 'react-native-vector-icons/Entypo';
import { Container, Header, Body, Title, Content, Form, Label, Item, Button, Input, Text } from 'native-base';

const styles = StyleSheet.create({
  logoContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 100
  },
  form: {
    paddingBottom: 10
  },
  button: {
    margin: 10
  }
});

const LoginScreen = class extends React.Component {
  constructor(props) {
    super(props);

    this.navigation = props.navigation;

    this.checkLocalStorage();

    this.state = {
      user: null
    };
  }

  login() {
    this.navigation.dispatch({ type: 'Login', params: { user: this.state.user } });
  }

  checkLocalStorage() {
    AsyncStorage.getItem('@githubClient:user').then((user) => {
      if (user){
        this.setState({ user });
        this.login();
      }
    });
  }

  render() {
    return (
      <Container>
        <Header>
          <Body>
            <Title>Login</Title>
          </Body>
        </Header>
        <Content>
          <View style={styles.logoContainer}>
            <Entypo name='github' size={100} />
          </View>
          <Form style={styles.form}>
            <Item floatingLabel last>
              <Label>Username</Label>
              <Input onChangeText={(t) => this.setState({ user: t })}/>
            </Item>
          </Form>
          <Button block style={styles.button}
            onPress={() => this.login()}>
            <Text>GO</Text>
          </Button>
        </Content>
      </Container>
    );
  }
};

LoginScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

export default LoginScreen;