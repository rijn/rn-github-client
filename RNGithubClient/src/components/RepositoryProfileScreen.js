import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { StyleSheet, View, WebView } from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons';
import {
  Container, Content, Separator, ListItem, Text, Left, Body, Right, Thumbnail,
  Tab, Tabs, ScrollableTab, Icon
} from 'native-base';
import { Col, Row, Grid } from 'react-native-easy-grid'

import UserList from './UserList';

import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff'
  },
  icon: {
    width: 20
  },
  list: {
    marginTop: 5
  },
  inline: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'center'
  },
  inlineInnerCeil: {
    marginRight: 10
  },
  ceil: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
    height: 100,
    borderColor: '#c9c9c9'
  },
  ceilBottom: {
    borderBottomWidth: 0.5
  },
  ceilLeft: {
    borderRightWidth: 0.5
  }
});

const iconSize = 20;

/**
 * Attributes and Icon map
 */
const attrMap = {
  createdAt: {
    displayName: 'Created',
    icon: 'ios-calendar-outline',
    processor: v => new Date(v).toLocaleDateString()
  },
  owner: {
    displayName: 'Owner',
    icon: 'ios-contact-outline',
    processor: ({ login }) => login,
    clickable: true,
    onPress: ({ login }, dispatch) => { dispatch({ type: 'UserProfile', params: { user: { login } } }); }
  }
};

/**
 * Repository profile screen
 * @class RepositoryProfileScreen
 */
class RepositoryProfileScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  /**
   * Set the header based on navigation state
   */
  static navigationOptions = ({ navigation: { state: { params: { repository: { name, owner } } } } }) => ({
    title: `${owner}/${name}`,
  });

  /**
   * Primay language component
   * @param  {Object} primaryLanguage
   * @return {ReactDOM}
   */
  primaryLanguageView(primaryLanguage) {
    if (!primaryLanguage) return;
    let { name: primaryLanguageName, color: primaryLanguageColor } = primaryLanguage;
    return (
      <View style={[ styles.inline, styles.inlineInnerCeil ]}>
        <View style={{
          width: 7, height: 7, borderRadius: 7,
          marginRight: 5,
          backgroundColor: primaryLanguageColor
        }}></View>
        <Text note>{primaryLanguageName}</Text>
      </View>
    );
  }

  /**
   * Render function
   * @return {ReactDOM}
   */
  render() {
    let { repositoryProfile: { error, loading } } = this.props;

    if (error) {
      return <Text>{error.toString()}</Text>;
    } else if (loading) {
      return <Text>Loading</Text>;
    } else {

      let { repositoryProfile: { repository }, dispatch } = this.props;

      let Attrs = _.chain(repository).pick(_.keys(attrMap)).map((value, key) => {
        if (!value) return;
        return (
          <ListItem icon
            last={key === _.last(_.keys(attrMap))}
            key={key}
            onPress={attrMap[key].clickable &&
              (() => {
                return () => {
                  attrMap[key].onPress(value, dispatch);
                }
              })(value, dispatch)
            }
            >
            <Left>
              <Ionicons name={attrMap[key].icon} size={iconSize} style={styles.icon} />
            </Left>
            <Body>
              <Text>{attrMap[key].displayName}</Text>
            </Body>
            <Right>
              <Text>{attrMap[key].processor ? attrMap[key].processor(value) : _.identity(value)}</Text>
              {attrMap[key].clickable && <Icon name='arrow-forward' />}
            </Right>
          </ListItem>
        );
      }).value();

      return (
        <Container>
          <Content>
            <View style={styles.container}>
              <Separator bordered>
                <Text>Profile</Text>
              </Separator>
              <ListItem>
                <Grid>
                  <Row>
                    <Text>{repository.nameWithOwner}</Text>
                  </Row>
                  <Row>
                    <Text note style={styles.list}>{repository.description}</Text>
                  </Row>
                  <View style={[ styles.inline, { marginTop: 5 } ]}>
                    {this.primaryLanguageView(repository.primaryLanguage)}
                    <View style={[ styles.inline, styles.inlineInnerCeil ]}>
                      <Ionicons name='ios-star-outline' style={{ marginRight: 5 }} />
                      <Text note>{repository.stargazers.totalCount}</Text>
                    </View>
                    <View style={[ styles.inline, styles.inlineInnerCeil ]}>
                      <Ionicons name='ios-git-network-outline' style={{ marginRight: 5 }} />
                      <Text note>{repository.forks.totalCount}</Text>
                    </View>
                    <Text note>Updated at {new Date(repository.updatedAt).toLocaleDateString()}</Text>
                  </View>
                </Grid>
              </ListItem>
              {Attrs}
              <Separator bordered>
              </Separator>
              <ListItem last onPress={() => { dispatch({ type: 'WebView', params: { url: repository.url } }); }}>
                <Body>
                  <Text>View webpage</Text>
                </Body>
                <Right>
                  <Ionicons name='ios-arrow-forward' />
                </Right>
              </ListItem>
            </View>
          </Content>
        </Container>
      );
    }
  }
};

/**
 * Prop Types
 * Required repository profile information
 */
RepositoryProfileScreen.propTypes = {
  repositoryProfile: PropTypes.shape({
    loading: PropTypes.bool.isRequired,
    repository: PropTypes.object,
  }).isRequired
};

/**
 * Get repository profile query
 */
const GetRepositoryProfileQuery = gql`
  query getRepositoryProfile($name: String!, $owner: String!) {
    repository(name: $name, owner: $owner) {
      id
      name
      updatedAt
      createdAt
      description
      nameWithOwner
      url
      owner {
        id
        login
        avatarUrl
      }
      primaryLanguage {
        name
        color
      }
      stargazers {
        totalCount
      }
      forks {
        totalCount
      }
      watchers {
        totalCount
      }
      issues {
        totalCount
      }
      pullRequests {
        totalCount
      }
      isFork
      parent {
        nameWithOwner
      }
      licenseInfo {
        name
        nickname
      }
    }
  }
`;

/**
 * Apollo Container
 */
const withInfo = graphql(GetRepositoryProfileQuery, {
  name: 'repositoryProfile',
  options: (props) => ({
    pollInterval: 20000,
    variables: props.navigation.state.params.repository
  })
});

export default connect()(withInfo(RepositoryProfileScreen));