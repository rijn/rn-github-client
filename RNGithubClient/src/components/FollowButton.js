import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { StyleSheet, View, FlatList, Dimensions } from 'react-native';
import {
  Container, Content, Separator, List, ListItem, Text, Left, Body, Right,
  Thumbnail, Button, Spinner, Toast
} from 'native-base';

import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import {
  followUser as callFollowUser,
  unfollowUser as callUnfollowUser
} from '../services/githubFollow';

import ErrorToast from './ErrorToast';

const styles = StyleSheet.create({
});

class FollowButton extends React.Component {
  constructor(props) {
    super(props);
  }

  async followUser() {
    let { login } = this.props;
    try {
      const res = await callFollowUser(login);
      this.props.data.refetch();
    } catch (e) {
      ErrorToast(e);
    }
  }

  async unfollowUser() {
    let { login } = this.props;
    try {
      const res = await callUnfollowUser(login);
      this.props.data.refetch();
    } catch (e) {
      ErrorToast(e);
    }
  }

  render() {
    let { dispatch, data } = this.props;

    console.log(this.props);

    if (!data || data.loading) {
      return <Spinner color='#00a6de' />;
    }

    let { user: { viewerCanFollow, viewerIsFollowing } } = data;

    if (!viewerCanFollow) {
      return <View></View>
    }

    return (
      <Button small info bordered={viewerIsFollowing}
        onPress={() => { viewerIsFollowing ? this.unfollowUser() : this.followUser(); }}>
        <Text>{viewerIsFollowing ? 'Followed' : 'Follow'}</Text>
      </Button>
    );
  }
};

FollowButton.propTypes = {
  login: PropTypes.string.isRequired
};

const GetFollowStatusQuery = gql`
  query GetFollowStatus($login: String!) {
    user(login: $login) {
      viewerCanFollow
      viewerIsFollowing
    }
  }
`;

const withInfo = graphql(GetFollowStatusQuery, {
  options: (props) => ({
    variables: props
  })
});

export default connect()(withInfo(FollowButton));