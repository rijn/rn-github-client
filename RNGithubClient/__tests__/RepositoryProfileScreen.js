import 'react-native';
import React from 'react';
import { Provider } from 'react-redux';
import gql from 'graphql-tag';
import { mockServer, makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { graphql, ApolloProvider } from 'react-apollo';
import { ApolloClient, createNetworkInterface } from 'apollo-client';
import { mockNetworkInterfaceWithSchema } from 'apollo-test-utils';
import { RepositoryProfileScreen } from '../src/components/RepositoryProfileScreen';

import renderer from 'react-test-renderer';

import configureStore from 'redux-mock-store'

test('renders loading correctly', () => {
  const tree = renderer.create(
    <RepositoryProfileScreen repositoryProfile={{ loading: true }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders error correctly', () => {
  const tree = renderer.create(
    <RepositoryProfileScreen repositoryProfile={{ error: new Error('Error test'), loading: false }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders information correctly', () => {
  let typeDefs = `
      type Repository {
        viewerHasStarred: Boolean
      }

      type Query {
        repository(name: String!, owner: String!): Repository
      }
    `;
  let schema = makeExecutableSchema({ typeDefs });
  addMockFunctionsToSchema({ schema });
  let server = mockServer(schema, {
    User: () => ({ viewerCanFollow: true, viewerIsFollowing: false }),
  });
  let store = configureStore()({})
  let networkInterface = mockNetworkInterfaceWithSchema({ schema, server });
  let client = new ApolloClient({ networkInterface });
  let tree = renderer.create(
    <ApolloProvider store={store} client={client}>
      <RepositoryProfileScreen repositoryProfile={{
        error: null,
        loading: false,
        repository: {
          id: 'test',
          name: 'test',
          updatedAt: 'test',
          createdAt: 'test',
          description: 'test',
          nameWithOwner: 'test',
          url: 'test',
          owner: {
            id: 'test',
            login: 'test',
            avatarUrl: 'test'
          },
          primaryLanguage: {
            name: 'test',
            color: '#ffffff'
          },
          stargazers: { totalCount: 0 },
          forks: { totalCount: 0 },
          watchers: { totalCount: 0 },
          issues: { totalCount: 0 },
          pullRequests: { totalCount: 0 },
          isFork: false,
          parent: {
            nameWithOwner: 'test'
          },
          licenseInfo: {
            name: 'test',
            nickname: 'test'
          }
        }
      }} />
    </ApolloProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});