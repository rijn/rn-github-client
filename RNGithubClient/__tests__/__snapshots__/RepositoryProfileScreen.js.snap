// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#000",
      "fontFamily": "System",
      "fontSize": 16,
    }
  }
  uppercase={false}
  virtual={undefined}
>
  Error: Error test
</Text>
`;

exports[`renders information correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
          }
        }
      >
        <View
          bordered={true}
          style={
            Object {
              "backgroundColor": "#F0EFF5",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "borderTopWidth": 0.5,
              "flex": 1,
              "height": 35,
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingTop": 12,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#777",
                "fontFamily": "System",
                "fontSize": 12,
              }
            }
            uppercase={false}
            virtual={undefined}
          >
            Profile
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 15,
                "paddingBottom": 13,
                "paddingRight": 15,
                "paddingTop": 13,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "alignSelf": "flex-start",
                          },
                        ]
                      }
                      uppercase={false}
                      virtual={undefined}
                    >
                      test
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <ActivityIndicator
                      animating={true}
                      color="#00a6de"
                      hidesWhenStopped={true}
                      size="large"
                      style={
                        Object {
                          "height": 80,
                        }
                      }
                      virtual={undefined}
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  note={true}
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#808080",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "200",
                      },
                      Object {
                        "marginTop": 5,
                      },
                    ]
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  test
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "flexWrap": "wrap",
                    },
                    Object {
                      "marginTop": 5,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                      },
                      Object {
                        "marginRight": 10,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderRadius": 7,
                        "height": 7,
                        "marginRight": 5,
                        "width": 7,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    note={true}
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#808080",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "200",
                      }
                    }
                    uppercase={false}
                    virtual={undefined}
                  >
                    test
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                      },
                      Object {
                        "marginRight": 10,
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Object {
                          "marginRight": 5,
                        },
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    note={true}
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#808080",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "200",
                      }
                    }
                    uppercase={false}
                    virtual={undefined}
                  >
                    0
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                      },
                      Object {
                        "marginRight": 10,
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Object {
                          "marginRight": 5,
                        },
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    note={true}
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#808080",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "200",
                      }
                    }
                    uppercase={false}
                    virtual={undefined}
                  >
                    0
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  note={true}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#808080",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "200",
                    }
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  Updated at 
                  Invalid Date
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            icon={true}
            last={false}
            onPress={undefined}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": null,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "marginLeft": 15,
                "paddingBottom": null,
                "paddingRight": null,
                "paddingTop": null,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "center",
                  "paddingRight": 15,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "width": 20,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 17,
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                Created
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "flex-end",
                  "paddingBottom": null,
                  "paddingLeft": null,
                  "paddingRight": 15,
                  "paddingTop": null,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": null,
                    "color": "#8F8E95",
                    "fontFamily": "System",
                    "fontSize": 17,
                    "textAlign": "center",
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                Invalid Date
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            icon={true}
            last={true}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "marginLeft": -15,
                "paddingBottom": null,
                "paddingLeft": 30,
                "paddingRight": null,
                "paddingTop": null,
                "top": 1,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "center",
                  "paddingRight": 15,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "width": 20,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": null,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 17,
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                Owner
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderBottomWidth": null,
                  "borderColor": "#c9c9c9",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "flex-end",
                  "paddingBottom": null,
                  "paddingLeft": null,
                  "paddingRight": 15,
                  "paddingTop": null,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": null,
                    "color": "#8F8E95",
                    "fontFamily": "System",
                    "fontSize": 17,
                    "textAlign": "center",
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                test
              </Text>
              <Text
                accessible={true}
                allowFontScaling={false}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 30,
                      },
                      Object {
                        "alignSelf": "center",
                        "color": "#C8C7CC",
                        "fontSize": 20,
                        "paddingLeft": 10,
                        "paddingTop": 3,
                      },
                    ],
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          bordered={true}
          style={
            Object {
              "backgroundColor": "#F0EFF5",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "borderTopWidth": 0.5,
              "flex": 1,
              "height": 35,
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingTop": 12,
            }
          }
          virtual={undefined}
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            last={true}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": -15,
                "paddingBottom": 13,
                "paddingLeft": 30,
                "paddingRight": 15,
                "paddingTop": 13,
                "top": 1,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "flex": 1,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                View webpage
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "alignSelf": "center",
                  "flex": 0.28,
                  "paddingBottom": null,
                  "paddingLeft": null,
                  "paddingRight": null,
                  "paddingTop": null,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders loading correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#00a6de"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "height": 80,
    }
  }
  virtual={undefined}
/>
`;
